test/compress/collapse_vars.js

```diff
@@ -2256,3 +2256,31 @@ issue_2187_3: {
     }
     expect_stdout: "1"
 }
+
+issue_2203_1: {
+    options = {
+        collapse_vars: true,
+        unused: true,
+    }
+    input: {
+        console.log({
+            a: 42,
+            b: function() {
+                return function(c) {
+                    return c.a;
+                }(this);
```

Comment:
Great.

Could you make a test that's a copy of `issue_2203_1` but replace `this` with `global`? It should have the same result, but that argument be inlined.